### ISSUE TITLE
[Bug] Approximate location on Android 12 doesn't work correct.

### ISSUE THREAD

**Issue by orikworld** [xamarin/Essentials#1972]
### Description
      Can not get approximate location on android 12 because:
      line (Permissions.EnsureGrantedAsync<Permissions.LocationWhenInUse>())  expected only permissions granted:
  https://github.com/xamarin/Essentials/blob/72dda322aaa583873eb517b44fdcba3e87e5f35f/Xamarin.Essentials/Geolocation/Geolocation.android.cs#L40

### Steps to Reproduce

1. Select approximate location in popup
2. Try to get permissions for locationInUse => Restricted 
3. Try to get location => inner exception

### Expected Behavior
     Geolocation gets approximate location on android 12
### Actual Behavior
    Geolocation throw exception
 ### Basic Information

- Version with issue:
- Platform Target Frameworks: 
  - Android:

**Comment by SharpMobileCode** [comment]
I'm seeing the same issue as you when users select "Approximate" location.  The permission status comes back as Restricted, which is expected.  But when we perform a `Geolocation.GetLocationAsync(request)`, a null result comes back instead of a `Location`.  

I'm just taking a guess here looking at the code, but I think this line:

https://github.com/xamarin/Essentials/blob/72dda322aaa583873eb517b44fdcba3e87e5f35f/Xamarin.Essentials/Permissions/Permissions.shared.cs#L28

might need to be checking for `PermissionStatus.Restricted` as well?

Current Code
```c#
if (status != PermissionStatus.Granted)
    throw new PermissionException($"{typeof(TPermission).Name} permission was not granted: {status}");
```

Proposed Change?
```c#
if (status != PermissionStatus.Granted && status != PermissionStatus.Restricted)
    throw new PermissionException($"{typeof(TPermission).Name} permission was not granted: {status}");
```

Thoughts anyone?

cc: @jfversluis

**Comment by jfversluis** [comment]
@aritchie I guess this might be something that came from #1937 ?

**Comment by jfversluis** [comment]
@SharpMobileCode I see you have a analysis and proposal already, would you be willing and able to put it in a PR? You should be able to then also test if it actually fixes it for you :)

**Comment by aritchie** [comment]
@jfversluis yup.  I only changed the permission requests, not any of the gatekeeper methods.  I just sent a PR for this: https://github.com/xamarin/Essentials/pull/1976

I would appreciate if the users in this issue test it please.

**Comment by jfversluis** [comment]
Let me know if you got around to this @SharpMobileCode thanks!

**Comment by SharpMobileCode** [comment]
Okay, I've got it integrated into my app now (using project references instead) and it doesn't appear to work as expected.  Currently running through the debugger trying to figure some stuff out.  Thanks!

**Comment by SharpMobileCode** [comment]
Okay, now it seems that when the user selects "Approximate Location", the `locationManager.RequestUpdates(provider, 0, 0, listener, looper);` call does not seem to execute the `listener`.  This causes

https://github.com/xamarin/Essentials/blob/c0ea6b0fdcd9002f00407c4e2815f6ae17c8e0d6/Xamarin.Essentials/Geolocation/Geolocation.android.cs#L84

to `await` indefinitely because:

https://github.com/xamarin/Essentials/blob/c0ea6b0fdcd9002f00407c4e2815f6ae17c8e0d6/Xamarin.Essentials/Geolocation/Geolocation.android.cs#L91

is never invoked from the listener.  So it just kind of freezes there.  However, if the user selects "Precise" location, then everything works as expected.  

I've only ever used `FusedLocationProviderClient` natively for geolocation, so I'm not too familiar with `LocationManager`.  I'll have to research this further.  Any insight or experiences regarding this?

Thanks!

**Comment by jfversluis** [comment]
Thanks for working together on this with us @SharpMobileCode ! I'll look into this asap as well :)
